test/parallel/test-http2-trailers.js

```diff
@@ -18,6 +18,7 @@ server.on('stream', common.mustCall(onStream));
 function onStream(stream, headers, flags) {
   stream.on('trailers', common.mustCall((headers) => {
     assert.strictEqual(headers[trailerKey], trailerValue);
+    stream.end(body);
```

Comment:
Is this needed? I do not think this is the API we should look for in this case. If we are piping a stream to (`origin.pipe(dest)` or `pipeline(origin, dest)`), that stream will end by default. Maybe we can wait with a `_final` handler?

Reply:
@mcollina If a user pipes a stream and have not tried to process incoming data then there's no way we can both close the stream properly and let them get the trailers.

Reply:
The alternative to how this test is modified is to call `resume` or bind a data handler.

Reply:
This needs to be documented.

Reply:
> This needs to be documented.

@mcollina Do you mean a comment needs to be added to the test? Or something needs to be added to the `http2` documentation?

Reply:
I think this should be documented in the documentation.

Reply:
I agree. Once that's added, do you approve for this to land @mcollina?

Reply:
Yes, but I think trailers is something that we should go back to.

Reply:
@mcollina I think other than documenting we can't do much. The flow can go something like this:

1. The third party sends some data (perhaps it's even larger than overall `windowSize`)
2. We don't handle that data, so the third party never sends the rest of the data nor the trailers
3. We send a response and initiate closing the connection since we haven't shown any interest in consuming the incoming data.

With `http` that would look different because data is just written to the socket immediately and all of it can come through. There's no mechanism there to tell the other side not to write the full payload along with the trailers.

Reply:
Ok